ada_2_fhir-r4/zibs2020/payload/0.0/nl-core-TobaccoUse.xsl

```diff
@@ -0,0 +1,182 @@
+                                <code value="53661000146106"/>
+                                <display value="Type of tobacco used"/>
+                            </coding>
+                        </code>
+                        <valueCodeableConcept>
+                            <xsl:call-template name="code-to-CodeableConcept"/>
+                        </valueCodeableConcept>
+                    </component>
+                </xsl:for-each>
+                
+                <xsl:for-each select="waarneming_gebruik/hoeveelheid">
+                    <component>
+                        <code>
+                            <coding>
+                                <system value="http://snomed.info/sct"/>
+                                <code value="266918002"/>
+                                <display value="Tobacco smoking consumption"/>
+                            </coding>
+                        </code>
+                        <valueQuantity>
+                            <xsl:call-template name="hoeveelheid-to-Quantity">
+                                <xsl:with-param name="in" select="."/>
+                            </xsl:call-template>
+                        </valueQuantity	>
+                    </component>
+                </xsl:for-each>
+                
+                <xsl:for-each select="waarneming_gebruik/pack_years">
+                    <component>
+                        <code>
+                            <coding>
+                                <system value="http://snomed.info/sct"/>
+                                <code value="401201003"/>
+                                <display value="pakjaren"/>
+                            </coding>
+                        </code>
+                        <valueInteger>
+                            <xsl:attribute name="value" select="./@value"/>
+                        </valueInteger>
+                    </component>
+                </xsl:for-each>
+            </Observation>
+        </xsl:for-each>
+    </xsl:template>
+    
+    <xd:doc>
+        <xd:desc>Template to generate a unique id to identify this instance.</xd:desc>
+    </xd:doc>
+    <xsl:template match="tabak_gebruik" mode="_generateId">
+        <xsl:variable name="parts">
+            <xsl:text>TobaccoUse</xsl:text>
+            <xsl:value-of select="tabak_gebruik_status/@displayName"/>
+            <xsl:value-of select="soort_tabak_gebruik/@displayName"/>
+            <xsl:value-of select="waarneming_gebruik/start_datum/@value"/>
+            <xsl:value-of select="waarneming_gebruik/stop_datum/@value"/>
+            <xsl:value-of select="waarneming_gebruik/hoeveelheid/@value"/>
+            <xsl:value-of select="waarneming_gebruik/pack_years/@value"/>
+            <xsl:value-of select="toelichting/@value"/>
+        </xsl:variable>
+        <xsl:value-of select="substring(replace(string-join($parts, '-'), '[^A-Za-z0-9-.]', ''), 1, 64)"/>
+    </xsl:template>
```

Comment:
_generateId hadden we nog niet echt algemene afspraken over (mist bij nog heel veel zibs, omdat de fallback vaak voldoende is). Ik vind dit wat heftig veel parts, misschien even afspraken over maken.

Reply:
Ik heb eigenlijk geen idee of dat wat uit maakt? Maar hier afspraken overmaken lijkt me meer dan goed. Als je minder parts moet selecteren, welke kies je dan?

Reply:
Ja wat is nou een goede ID. Ik zou het niet weten.

Maar goed, blijft een beetje hypothetisch voor dit project, want we hebben hier in _driverInclude het overruled dat alles nl-core-Nogwat wordt. Heeft dus nog niet echt prioriteit.